core/src/com/unciv/ui/utils/ImageGetter.kt

```diff
@@ -67,6 +67,51 @@ object ImageGetter {
         }
     }
 
+
+    /**
+     * Colors a multilayer image and returns it as a list of layers (Image).
+     *
+     * @param   baseFileName    The filename of the base image.
+     *                              For example: TileSets/FantasyHex/Units/Warrior
+     *
+     * @param   colors          The list of colors, one per layer. No coloring is applied to layers
+     *                              whose color is null.
+     *
+     * @return  The list of layers colored. The layers are sorted by alphabetical order and colors
+     *              are applied, one per layer, in the same order. If a color is null, no coloring is
+     *              performed on such layer (it stays as it is). If there are less colors than layers,
+     *              the last layers are not colored. Defaults to an empty list if there is no layer
+     *              corresponding to baseFileName.
+     *
+     * Example:
+     *      getLayeredImageColored("TileSets/FantasyHex/Units/Warrior", null, Color.GOLD, Color.RED)
+     *
+     *      All images in the atlas that match the pattern "TileSets/FantasyHex/Units/Warrior" or
+     *      "TileSets/FantasyHex/Units/Warrior-NUMBER" are retrieved alphabetically:
+     *          - TileSets/FantasyHex/Units/Warrior
+     *          - TileSets/FantasyHex/Units/Warrior-1
+     *          - TileSets/FantasyHex/Units/Warrior-2983
+     *
+     *      It returns a list in which first layer has unmodified colors, the second is colored in
+     *      GOLD and the third in RED.
+     */
+    fun getLayeredImageColored(baseFileName: String, vararg colors: Color?): ArrayList<Image> {
+        val regex = Regex("^$baseFileName(?>-\\d+)?$")
```

Comment:
Regexes are notoriously heavy lifting. Running a regex on EVERY unit tile (when creating the map, which is every turn) means hundreds of regex checks per tile, times hundreds of tiles.
Another problem is that in the case that we go up to 11 (hard to imagine but let's roll with it), 11 is alphabetically between 1 and 2.

Instead, lets do like
i=1
while(textureRegionDrawables.containsKey("$baseFileName-$i"){
 ...
 i++
}

This DOES mean that if you're missing, say, 2 and you have 3, it won't be rendered. But I think that's fair.

Alternatively, we could have them hard-named as $baseFileName, $baseFileName-innerColor, $baseFileName-outerColor

Reply:
Good points! I believe regex performances relies on patterns complexity, anyway it is for sure much more performant to look for an exact key. I think modders and artists would like to have as much freedom as possible, so I'm going with the loop solution.